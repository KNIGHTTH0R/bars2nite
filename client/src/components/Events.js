import React, { Component } from 'react';
import { connect } from 'react-redux';
import { onLoadUserReservedBars } from '../store/actions';
import EventItem from '../helpers/EventItem';

export class Events extends Component {
  componentDidMount() {
    this.props.onLoadUserReservedBars();
  }

  render() {
    let eventContent;
    if (this.props.userBars) {
      eventContent = this.props.userBars.map(bar => {
        return (
          <EventItem
            key={bar.name}
            name={bar.name}
            going={bar.numberGoing}
            image={bar.image}
            yelpId={bar.yelpId}
            location={bar.location}
            website={bar.website}
            price={bar.price}
          />
        );
      });
    }

    let content;
    if (this.props.loading) {
      content = <div className="loader">Loading...</div>;
    } else {
      content = (
        <div>
          <h2 className="Event__heading">Your Events</h2>
          <div className="Event__content">{eventContent}</div>
        </div>
      );
    }

    return <div className="Event">{content}</div>;
  }
}

const mapStateToProps = state => {
  return {
    userBars: state.bars.userBars,
    loading: state.bars.loading
  };
};

export default connect(
  mapStateToProps,
  { onLoadUserReservedBars }
)(Events);
