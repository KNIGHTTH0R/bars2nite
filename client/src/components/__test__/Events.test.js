import React from 'react';
import { shallow, mount } from 'enzyme';
import { Events } from '../Events';

describe('Events', () => {
  const mockOnLoadUserReservedBars = jest.fn();

  const props = {
    onLoadUserReservedBars: mockOnLoadUserReservedBars
  };

  let events = shallow(<Events {...props} />);

  it('renders properly', () => {
    expect(events).toMatchSnapshot();
  });

  describe('when mounted', () => {
    beforeEach(() => {
      events = mount(<Events {...props} />);
    });

    it('renders the page title', () => {
      expect(events.find('.Event__heading').text()).toEqual('Your Events');
    });

    it('dispatches the `onLoadUserReservedBars` it receives from props', () => {
      expect(mockOnLoadUserReservedBars).toHaveBeenCalled();
    });
  });
});
