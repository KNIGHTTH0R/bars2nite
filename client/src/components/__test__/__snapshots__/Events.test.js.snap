// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Events
    onLoadUserReservedBars={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h2
                className="Event__heading"
        >
                Your Events
        </h2>
        <div
                className="Event__content"
        />
</div>,
      "className": "Event",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="Event__heading"
>
            Your Events
</h2>,
          <div
            className="Event__content"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your Events",
            "className": "Event__heading",
          },
          "ref": null,
          "rendered": "Your Events",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "Event__content",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h2
                    className="Event__heading"
          >
                    Your Events
          </h2>
          <div
                    className="Event__content"
          />
</div>,
        "className": "Event",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="Event__heading"
>
              Your Events
</h2>,
            <div
              className="Event__content"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your Events",
              "className": "Event__heading",
            },
            "ref": null,
            "rendered": "Your Events",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "Event__content",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
